### ISSUE TITLE
A Varity of Hyperdrives for Different Factions/Species

### ISSUE THREAD

**Issue by CaptianGrim** [endless-sky/endless-sky#8940]
### Problem Description

I have been playing this game for a while now. Played almost entire story line or Grinding for hours to get Heavy Warships.
For every factions or aliens in the game they have a unique ships,weapons,turrets,hand to hand weapons,reactors etc...
except for one particular outfit.. the **Hyperdrive**. All aliens and human are using the same **Hyperdrive** (except for the ones who possess the jumpdrives). At first it did'nt bothered me but after playing the story line i think its better to have slight variations of hyperdrives for different alien factions to make the game for interesting.

### Related Issue Links

 New hyperdrive inspiration #7067 
 Alien jump drives #8721 

### Desired Solution

To start all aliens factions have at least some variations of the **Hyperdrive**. This would make the game a little for interesting.I propose some additional functionality to the **Hyperdrives**.
1. A new variability factor to the **Hyperdrive** "Jump Capability" this will determine the jump capability of the **Hyperdrive** if the Ship is too massive or mass of the ship is above a certain threshold a single **Hyperdrive** will not be able to jump to another star system

2. All aliens factions have different **Hyperdrives** so the "Jump Capability" factor of the Drive will be different for different aliens.

3. Add the "Heat Generated" to the **Hyperdrives** when the ship to about the jump into another star system its **Hyperdrive** will activate along with it some heat will also be generated. And if a ships heat value is high due to weapon firing or some other factors and when they use their **Hyperdrive** at such circumstances the additional heat produced will disable their ships due to overheating. 

4. The **Hyperdrive** will not activate if a ships heat level is above a certain threshold

5. If a ship receive too much cargo it cannot jump into another system due to high mass.
solutions: eject some cargo to lighten the ship or install more **Hyperdrives** to increase the max "jump mass" of the ship

6. When jumping from one starsystem to another it will some time for the **Hyperdrive** to reactivate a **"Cooldown Period**".
The "**Cooldown Period**" will only be about 3 to 4 seconds. This will be different for different **Hyperdives** of the aliens factions

### Alternative Approaches

Alternative Approach:

you can make a storyline on how Hyperdrives came it existence, and how most aliens factions have same hyperdrives

**Comment by lumbar527** [comment]
Related (but for Jump Drives): #8721

**Comment by CaptianGrim** [comment]
> Related (but for Jump Drives): #8721

ohh thanks:)

**Comment by Saugia** [comment]
TL;DR, through discussions in the past, we don't want to have Jump Drives that do different things, and likely not Hyperdrives. 

A lot of the content in the game is based around how both the HD and JD work as they currently do, and oversaturating every faction with a special type of drive starts to get messy. We're going to keep "Hyperdrive", "Scram Drive" and "Jump Drive"'s universal. The Xapleaux for example will have a "String Drive", but that's something special to them, and not a strict variant of the HD or JD. 

There has been talk to have different *looking* hyperdrives, but as mentioned in #8721, we want outfit variants functionality first.
